test/lighthouse/LighthouseStringifyExtension.test.ts

```diff
@@ -0,0 +1,176 @@
+/**
+    Copyright 2022 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+import snapshot from 'snap-shot-it';
+
+import { stringify } from '../../src/stringify.js';
+import { LighthouseStringifyExtension } from '../../src/lighthouse/LighthouseStringifyExtension.js';
+import { UserFlow } from '../../src/Schema.js';
+
+describe('LighthouseStringifyExtension', () => {
+  it('handles ending timespan', async () => {
+    const flowJson: UserFlow = {
+      title: 'Test Flow',
+      steps: [
+        {
+          type: 'setViewport',
+          width: 757,
+          height: 988,
+          deviceScaleFactor: 3,
+          isMobile: true,
+          hasTouch: true,
+          isLandscape: false,
+        },
+        {
+          type: 'navigate',
+          url: 'https://example.com',
+          assertedEvents: [
+            {
+              type: 'navigation',
+              url: 'https://example.com',
+              title: '',
+            },
+          ],
+        },
+        {
+          type: 'click',
+          target: 'main',
+          selectors: [['#button']],
+          offsetY: 13.5625,
+          offsetX: 61,
+        },
+      ],
+    };
+
+    const scriptContents = await stringify(flowJson, {
+      extension: new LighthouseStringifyExtension(),
+    });
+
+    // Trim the output to the relevant stuff
```

Comment:
should we extract a helper function for trimming? Also, perhaps, it's better to search for `await browser.close()` to avoid the dangling `await` in the snapshot?